scanpointgenerator/core/dimension.py

```diff
@@ -51,6 +51,37 @@ def get_positions(self, axis):
         # scale up points for axis
         gen = [g for g in self.generators if axis in g.axes][0]
         points = gen.positions[axis]
+        return self.scale_and_index_points(gen, points)
```

Comment:
Is get_positions(axis) now just gen.positions[axis][get_mesh_map(axis)]?

Reply:
Yes it would be. It's slightly less efficient to do that though as it does a few things twice that just calling scale_and_index_points avoids. Can change it to that though if we're not too worried about that here

Reply:
Less code is probably better here, it's not much of a performance hit...

Reply:
ok, should look simpler now